src/modules/link/graphql/mutation/link_create.rs

```diff
@@ -48,6 +50,17 @@ impl LinkCreate {
         }
 
         let hash: String = if let Some(custom_hash) = input.custom_hash {
+            // validate hash is empty
+            if custom_hash.trim().is_empty() {
+                return Ok(Self {
+                    link: None,
+                    error: Some(LinkError {
+                        code: LinkErrorCode::CustomHashUsed,
```

Comment:
I suggest a dedicated error code here.

```suggestion
                        code: LinkErrorCode::InvalidCustomHash,
```

or perhaps `EmptyCustomHash`.